adapters/repos/db/vector/hnsw/deserializer.go

```diff
@@ -195,6 +198,15 @@ func (d *Deserializer) ReadNode(r io.Reader, res *DeserializationResult) error {
 		return err
 	}
 
+	// If the id is beyond maxNodeID, it is probably invalid (e.g. corrupt commit log or snapshot).
+	// Log the id and ignore it.
+	if id > maxNodeID {
+		d.logger.WithField("action", "hnsw_deserialization").
+			WithField("node_id", id).
```

Comment:
can you add some more infos here? :) Collection+shard+vector name would be great